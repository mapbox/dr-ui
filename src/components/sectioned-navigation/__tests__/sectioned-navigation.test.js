import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './sectioned-navigation-test-cases.js';

describe('sectioned-navigation', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.minimal.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.minimal;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.headingsOnly.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.headingsOnly;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.filter.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.filter;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
