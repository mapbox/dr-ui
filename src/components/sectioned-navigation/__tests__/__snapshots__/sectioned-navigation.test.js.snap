// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sectioned-navigation With a title, linked section headings, and counts renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div
    className="txt-l color-blue txt-fancy mb12 block-mm none"
  >
    Examples
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#getting-started"
      >
        Getting started (3)
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#foo"
        >
          Camera animation
        </a>
        <a
          className="color-blue-on-hover block mb6 txt-bold"
          href="#footoo"
        >
          Mark a place
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooandyou"
        >
          Apply a style
        </a>
      </div>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#markers"
      >
        Markers and callouts (2)
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooboo"
        >
          Annotation models
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#foocrew"
        >
          Callouts
        </a>
      </div>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#getting-started-again"
      >
        Getting started again (1)
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooblue"
        >
          Apply a style
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`sectioned-navigation With a title, linked section headings, counts, and filter renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div
    className="txt-l color-blue txt-fancy mb12 block-mm none"
  >
    Examples
  </div>
  <div
    className="mb18"
  >
    <div
      className="relative bg-white"
    >
      <input
        aria-required={true}
        className="input"
        data-test="filter-input"
        id="filter"
        name="filter"
        onChange={[Function]}
        placeholder="Filter list"
        type="text"
        value=""
      />
      <div
        role="alert"
      />
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#getting-started"
      >
        Getting started (3)
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#foo"
        >
          Camera animation
        </a>
        <a
          className="color-blue-on-hover block mb6 txt-bold"
          href="#footoo"
        >
          Mark a place
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooandyou"
        >
          Apply a style
        </a>
      </div>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#markers"
      >
        Markers and callouts (2)
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooboo"
        >
          Annotation models
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#foocrew"
        >
          Callouts
        </a>
      </div>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#getting-started-again"
      >
        Getting started again (1)
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooblue"
        >
          Apply a style
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`sectioned-navigation Without title, without linked section headings, without counts renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div>
    <div>
      <div
        className="block txt-bold color-gray py12"
      >
        Getting started
      </div>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#footwo"
        >
          Camera animation
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#foo-to"
        >
          Mark a place
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#footoo"
        >
          Apply a style
        </a>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div
        className="block txt-bold color-gray py12"
      >
        Markers and callouts
      </div>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#foot"
        >
          Annotation models
        </a>
        <a
          className="color-blue-on-hover block mb6"
          href="#fooz"
        >
          Callouts
        </a>
      </div>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py12 color-blue-on-hover"
        href="#getting-started-again"
      >
        Getting started again
      </a>
      <div>
        <a
          className="color-blue-on-hover block mb6"
          href="#foobles"
        >
          Apply a style
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`sectioned-navigation Without title, without sub items renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py6 color-blue-on-hover"
        href="#getting-started"
      >
        Getting started (3)
      </a>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py6 color-blue-on-hover"
        href="#markers-and-callouts"
      >
        Markers and callouts (2)
      </a>
    </div>
  </div>
  <div>
    <div>
      <a
        className="block txt-bold color-gray py6 color-blue-on-hover"
        href="#getting-started-again"
      >
        Getting started again (1)
      </a>
    </div>
  </div>
</div>
`;
